libraries/chain/include/eosio/chain/incremental_merkle.hpp

```diff
@@ -238,14 +243,17 @@ class incremental_merkle_impl {
          }
       }
 
-//   private:
-      uint64_t                         _node_count;
+   private:
+      friend struct fc::reflector<incremental_merkle_impl>;
+      uint64_t                         _node_count = 0;
       Container<DigestType, Args...>   _active_nodes;
 };
 
-typedef incremental_merkle_impl<digest_type>               incremental_merkle;
-typedef incremental_merkle_impl<digest_type,shared_vector> shared_incremental_merkle;
+typedef incremental_merkle_impl<digest_type, true>                incremental_canonical_merkle;
+typedef incremental_merkle_impl<digest_type, true, shared_vector> shared_incremental_canonical_merkle;
+typedef incremental_merkle_impl<digest_type>                      incremental_merkle_tree;
```

Comment:
why `_tree` at the end of this one... seems inconsistent to me.

Reply:
Ah, names are hard. Canonical is not very meaningful to me, I think I'd prefer the following, but totally fine if you leave as is. 
My remark was that I thought having the suffix `_tree` in the last one seemed inconsistent, but again no big deal.

`merkle` -> `bitflip_merkle` or `bf_merkle`
`incremental_merkle` -> `incremental_bitflip_merkle` or `incremental_bf_merkle` 
new one `incremental_noop_merkle` or `incremental_passthru_merkle`

Reply:
Renamed `incremental_canonical_merkle` to `incremental_canonical_merkle_tree`

Reply:
Thanks! Please update the comment at the top of the PR as well.